### PR TITLE
[FIX] account: remove readonly constraint

### PR DESCRIPTION
**Pull request by malb-odoo**
This commit will remove the readonly constraint on the journal when an account move has been posted.

task: 4028973

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/views/account_move_views.xml [hunk 1]
```diff
@@ -857,8 +857,7 @@
                                      invisible="context.get('default_journal_id') and context.get('move_type', 'entry') != 'entry'"
                                      groups="account.group_account_readonly">
                                     <field name="journal_id"
-                                           options="{'no_create': True}"
-                                           attrs="{'readonly': [('posted_before', '=', True)]}"/>
+                                           options="{'no_create': True}"/>
```
Comment (william-andre) [inline addons/account/views/account_move_views.xml:860]: Only if there is no sequence number (`False`, `''` or `'/'`)

**Review by william-andre — CHANGES_REQUESTED** [review]
We need the user to be sure that they made a gap by forcing them to void the sequence number fields

File: addons/l10n_ar/tests/common.py [hunk 2]
```diff
@@ -747,4 +747,3 @@ def _set_today_rate(self, currency, value):
             rate.rate = value
         else:
             rate_obj.create({'company_id': self.env.company.id, 'currency_id': currency.id, 'rate': value})
-
```
Comment (malb-odoo) [inline addons/l10n_ar/tests/common.py:750]: pycharm doing his thing :eyes:

Comment (william-andre) [reply addons/l10n_ar/tests/common.py:750]: You can just remove all diff on that file though

File: addons/account/tests/test_account_move_out_invoice.py [hunk 3]
```diff
@@ -3294,6 +3294,8 @@ def _check_invoice_values(invoice):
         _check_invoice_values(invoice_create)
 
         move_form = Form(self.env['account.move'].with_context(default_move_type='out_invoice'))
+        # Allow the journal to be editable
+        move_form.name = False
```
Comment (william-andre) [inline addons/account/tests/test_account_move_out_invoice.py:3298]: Why is this needed for a change in the view?

Comment (malb-odoo) [reply addons/account/tests/test_account_move_out_invoice.py:3298]: They emulate the form view so the read only on the view matter

File: addons/l10n_ar/tests/common.py (hunk 2 above)
Comment (malb-odoo) [reply addons/l10n_ar/tests/common.py:750]: i need to remove my automatic end of file thingy, i will do it

Comment (william-andre) [reply addons/l10n_ar/tests/common.py]: You could also just have reset the file with git :wink: 
i.e. `git checkout HEAD~ -- addons/l10n_ar_tests/common.py`

File: addons/account/tests/test_account_move_out_invoice.py (hunk 3 above)
Comment (william-andre) [reply addons/account/tests/test_account_move_out_invoice.py:3298]: Oh, I see.
But it should still be editable when `posted_before = False` because you cannot really create a gap in that case; only make the sequence start at 2 instead of 1 (which is not an issue in most countries)

File: addons/l10n_ar/views/account_move_view.xml [hunk 4]
```diff
@@ -30,6 +30,10 @@
                     </div>
                 </group>
             </group>
+
+            <xpath expr="//div[@name='journal_div']//field[@name='journal_id']" position="attributes">
+                 <attribute name="attrs">{'readonly': [('posted_before', '=', True)]}</attribute>
+            </xpath>
```
Comment (william-andre) [inline addons/l10n_ar/views/account_move_view.xml:36]: This is far from ideal because it affects all the companies, not only the ones in Argentina.

File: addons/account/tests/test_account_move_out_invoice.py (hunk 3 above)
Comment (malb-odoo) [reply addons/account/tests/test_account_move_out_invoice.py:3298]: Do we need to change the domain then ? not sure if you are asking for a change ahah

Comment (william-andre) [reply addons/account/tests/test_account_move_out_invoice.py:3298]: yes

File: addons/l10n_ar/views/account_move_view.xml [hunk 5]
```diff
@@ -30,6 +30,10 @@
                     </div>
                 </group>
             </group>
+
+            <xpath expr="//div[@name='journal_div']//field[@name='journal_id']" position="attributes">
+                 <attribute name="attrs">{'readonly': ['|', '&amp;', ('country_code', '=', 'AR'), ('posted_before', '=', True), '&amp;', ('posted_before', '=', True), ('name', 'not in', (False, '', '/'))]}</attribute>
```
[the remark this answers is not in the document]
Comment (malb-odoo) [reply addons/l10n_ar/views/account_move_view.xml:35]: Had a test that was failing "test_01_create_invoice", this test is failing because when creating the invoice in the setup from the method _create_test_invoices_like_demo, the test_invoice_6 is using a different journal "sale_expo_journal_ri". 
So the test fail because the afip_pos_number is not the right one, the problem is that the afip_pos_number is changed in the _inverse_l10n_latam_document_number which is compute from the name and the name is compute from another field and this field is computed from another thing, from what i've seen the journal at a certain moment is not the right one and so it goes through the if statement to modify the pos_number. 

We came to the conclusion that putting an override for that case was ok but i can still try to find another way to fix the problem but it's a inverse/compute nightmare in there and we couldn't find the a cleaner solution ahah

[the remark this answers is not in the document]
Comment (malb-odoo) [reply addons/l10n_ar/views/account_move_view.xml:35]: Wait just relaunched the test and it's working :facepalm:

[the remark this answers is not in the document]
Comment (malb-odoo) [reply addons/l10n_ar/views/account_move_view.xml]: I will wait for the l10n build to end before pinging you for the r+, let's see if it's green ahah

**Review by william-andre — APPROVED** [review]
@robodoo r+
